workflow/react-project/src/components/Header/index.jsx

```diff
@@ -25,10 +32,34 @@ const Header = () => {
         <FontAwesomeIcon icon={faMartiniGlassCitrus} className="logo" />
         <p className="logo-title">Cocktail App</p>
       </div>
-      <Button onClick={() => handleModalStateChange('Authentication')}>Get Started</Button>
-      {isAuthenticationModalOpened && (
-        <Modal title="Authentication" setModalClosed={() => handleModalStateChange(null)} />
+      {!authenticationToken ? (
+        <Button onClick={() => handleModalStateChange('Authentication')}>Get Started</Button>
+      ) : (
+        <div className="header-icons-wrapper">
+          <FontAwesomeIcon icon={faMagnifyingGlass} className="header-icon" />
+          <FontAwesomeIcon icon={faStar} className="header-icon" />
+          <FontAwesomeIcon icon={faHouseUser} className="header-icon" />
+        </div>
       )}
+
+      {isAuthenticationModalOpened &&
+        (isSignInModalOpened ? (
+          <SignIn
+            title="Authentication"
+            setModalClosed={() => handleModalStateChange(null)}
+            setSignInModalOpened={() => {
+              setSignInModalOpened(!isSignInModalOpened);
+            }}
+          />
+        ) : (
+          <SignUp
+            title="Authentication"
+            setModalClosed={() => handleModalStateChange(null)}
+            setSignInModalOpened={() => {
+              setSignInModalOpened(!isSignInModalOpened);
+            }}
+          />
```

Comment:
look, this `SignIn`/`SignUp` modal thing is deal of Authentication modal only.
Header should open/close only `AuthenticationModal`, what is inside it have to be controlled by `AuthenticationModal`